src/world_generation/world_generator.rs

```diff
@@ -21,63 +22,108 @@ impl WorldGenerator {
     /// Creates a simple world with hills
     pub fn create_new_random_world(n_chunks: i32) -> World {
         //let mut noise = PerlinNoise::new(121, 32.);
-        let seed: u64 = 42;
-        let mut noise = MultiscalePerlinNoise::new(seed, BIOMES[0].noise_config.clone());
+        let seed: u64 = 10;
+        let mut noise: MultiscalePerlinNoise = MultiscalePerlinNoise::new(seed, BIOMES[0].noise_config.clone());
 
         let s = CHUNK_SIZE as f32;
         let mut chunks = vec![];
+        
+        let mut possible_tree_pos: Vec<[i32; 3]> = vec![];
 
         // Yes this is slow, but it will be fine for now
         for i in -n_chunks..n_chunks + 1 {
             for j in -n_chunks..n_chunks + 1 {
                 let x0 = i as f32 * s;
                 let z0 = j as f32 * s;
-                let mut chunk = Chunk::new([x0, z0]);
-
-                // get the height from the perlin noise for each block
-                for x in 0..8 {
-                    for z in 0..8 {
-                        let biome_t: u64 = BiomeGenerator::find_closest_biome(seed, x + x0 as i32, z + z0 as i32);
-                        
-                        let biome_config = &BIOMES[biome_t as usize];
-
-                        if SINGLE_NOISE_CONFIG {
-                            noise.change_config(BASE_BIOME_CONFIG);
-                        } else {
-                            noise.change_config(biome_config.noise_config.clone());
-                        }
-
-                        let height = biome_config.terrain_offset
-                            + biome_config.terrain_scale
-                                * noise.at([i as f32 * s + x as f32, j as f32 * s + z as f32]);
-
-                        let cube_height = height.floor() as i32;
-
-                        for y in 0..cube_height {
-                            let block_at_height = biome_config.get_block_at(cube_height - y - 1);
-
-                            if let Some(block) = block_at_height {
-                                chunk.add_cube(
-                                    Vector3::new(
-                                        i as f32 * s + x as f32,
-                                        y as f32,
-                                        j as f32 * s + z as f32,
-                                    ),
-                                    block,
-                                    0,
-                                );
-                            }
-                        }
-                    }
-                }
+
+                let (chunk, mut new_trees) = WorldGenerator::create_random_chunk(seed, [x0, z0], &mut noise);
```

Comment:
To be fair, `new_trees` does seem right. Instead, `possible_tree_positions` is much better. Or maybe `candidate_tree_positions`

Why ? 
- Here, you are using a tuple for returning several values of different semantics. The best practice would be to encapsulte the information of the semantic by creating a new type. 
- It's true that it's a little bit 'too much' to create a new type just for 1 private function (*it will be private, see comment below*). 
- Since you decide to not create a type for this, then your naming must be **absolutely explicit**

Note:  if the function `create_random_chunk` was public, then it would not be acceptable to return a tuple like that. For reference, you would use this: https://doc.rust-lang.org/rust-by-example/generics/new_types.html